Update Debian installation instructions

pnax opened the pull request:
This PR:
* aligns locale generation with CLI installation document
* apply relevant changes from Centos section to Debian (inspired from https://github.com/zonemaster/zonemaster-backend/pull/696)
* install databases before configuring Zonemaster
* fix incorrect option
* fix database ownership

matsduf reviewed (dismissed):
It looks good.

matsduf commented on docs/Installation.md:
```diff
@@ -376,9 +384,6 @@ sudo -u postgres psql -f $(perl -MFile::ShareDir -le 'print File::ShareDir::dist
 #### 4.2.3 Instructions for SQLite (Debian)
 
-> **Note:** Zonemaster with SQLite backend is not yet considered stable and anyway
-> not meant for an installation with heavy load.
```
I guess we should have the same statement -- or no statement -- for all OS. Should the statement that "Zonemaster with SQLite backend is not yet considered stable and anyway not meant for an installation 
with heavy load" be put somewhere else or not at all?

pnax replied:
Good point. I removed it because this release brings a stable SQLite support. But indeed the second part of the note is still relevant.
I'll revert this specific commit and make a new PR to update this note in all sections.

pnax commented:
Rebased on top of `develop` and removed commit https://github.com/zonemaster/zonemaster-backend/pull/700/commits/cbb6ee624c4f5b2907f005ac6538116fcdf6e2e8
